contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,452 @@
+        int256 _amount);
+
+    event RedeemEther(address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    event RedeemNativeToken(address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    event RedeemExternalToken(address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    // A struct holding the data for a contribution proposal
+    struct ContributionProposal {
+        uint256 nativeTokenReward; // Reward asked in the native token of the organization.
+        int256 reputationChange; // Organization reputation reward requested.
+        uint256 ethReward;
+        IERC20 externalToken;
+        uint256 externalTokenReward;
+        address payable beneficiary;
+        uint256 executionTime;
+        uint256 nativeTokenRewardLeft;
+        uint256 reputationChangeLeft;
+        uint256 ethRewardLeft;
+        uint256 externalTokenRewardLeft;
+    }
+
+    modifier onlyRewarder() {
+        if (rewarder != address(0)) {
+            require(msg.sender == rewarder, "only rewarder allowed to redeem");
+        }
+        _;
+    }
+
+    mapping(bytes32=>ContributionProposal) public organizationProposals;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    address public rewarder;
+
+    /**
+    * @dev enables this contract to receive ethers
+    */
+    function() external payable {
+    }
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar to mint reputation from
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters.
+     * @param _rewarder an address which allowed to redeem the contribution.
+       if _rewarder is 0 this param is agnored.
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        address _rewarder
+    )
+    external
+    {
+        require(avatar == Avatar(0), "can be called only one time");
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        rewarder = _rewarder;
+    }
+
+    /**
+    * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        require(organizationProposals[_proposalId].executionTime == 0);
+        require(organizationProposals[_proposalId].beneficiary != address(0));
```

Comment:
mmm. this was taken from  the universal `ContributionReward executeProposal ` .
It might be here to prevent execution by mistake(due to a bug or bad actor)  of a votingMachine of uninitilize proposal.
we need to give it another thought before removing that here.